### PR TITLE
Implemented LETSENCRYPT_SINGLE_DOMAIN_CERTS environment variable

### PR DESCRIPTION
**Pull request by SilverFire**
This PR replaces #610. This one is based on the `master` branch.

### CODE REVIEW

**Comment by SilverFire** [comment]
It seems that you've asked your questions earlier, but submitted them only today :)

File: app/letsencrypt_service_data.tmpl [hunk 1]
```diff
@@ -1,17 +1,44 @@
-LETSENCRYPT_CONTAINERS=({{ range $hosts, $containers := groupBy $ "Env.LETSENCRYPT_HOST" }}{{ if trim $hosts }}{{ range $container := $containers }} '{{ printf "%.12s" $container.ID }}' {{ end }}{{ end }}{{ end }})
+LETSENCRYPT_CONTAINERS=(
+    {{ range $hosts, $containers := groupBy $ "Env.LETSENCRYPT_HOST" }}
+        {{ if trim $hosts }}
+            {{ range $container := $containers }}
+                {{ if parseBool (coalesce $container.Env.LETSENCRYPT_SINGLE_DOMAIN_CERTS "false") }}
```
[the remark this answers is not in the document]
Comment (SilverFire) [reply app/letsencrypt_service_data.tmpl:5]: Yes, it's a https://golang.org/pkg/strconv/#ParseBool function

> ParseBool returns the boolean value represented by the string. It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False. Any other value returns an error.

File: docs/Let's-Encrypt-and-ACME.md [hunk 2]
```diff
@@ -26,6 +26,21 @@ $ docker run --detach \
 
 Let's Encrypt has a limit of [100 domains per certificate](https://letsencrypt.org/fr/docs/rate-limits/), while Buypass limit is [15 domains per certificate](https://www.buypass.com/ssl/products/go-ssl-campaign).
 
+#### Separate certificate for each domain
+
+The example above will issue a single [SAN](https://www.digicert.com/subject-alternative-name.htm) certificate for all the listed in `LETSENCRYPT_HOST` domains. If you need to have a separate certificate for each of the domains, you can add the `LETSENCRYPT_SINGLE_DOMAIN_CERTS` environment variable.
```
[the remark this answers is not in the document]
Comment (SilverFire) [reply docs/Let's-Encrypt-and-ACME.md:31]: > The example is below, not above 😄

No, I've actually meant the `Multi-domains certificates` example above)

[the remark this answers is not in the document]
Comment (SilverFire) [reply docs/Let's-Encrypt-and-ACME.md:31]: > Don't forget to tell which value (true) the env var has to be set to, even if it's in the example below.

Done

[the remark this answers is not in the document]
Comment (buchdag) [reply docs/Let's-Encrypt-and-ACME.md]: > Done

Thanks.

> No, I've actually meant the `Multi-domains certificates` example above).
> And the mentioned SAN is also above the example above.

Oh ok ! I guess that make sense and we'll eventually change it if someone complains that it doesn't.

**Comment by buchdag** [comment]
> It seems that you've asked your questions earlier, but submitted them only today :)

Woops 😨 

LGTM, merging.

**Pull request merged**